_posts/2024-06-14-authz-openfga-introduction.md

```diff
@@ -0,0 +1,264 @@
+For Jane's scenario, we could define a test to verify that she has the correct `can_view` permissions for GlobeCorp's data and the `can_edit` permissions for EcoLife.
+
+Here’s an example of how such a test might look:
+
+```yaml
+name: demo
+model_file: ./model.fga
+# Sets up the initial state by defining specific relationships
+tuples:
+  - user: organization:GlobalCorp
+    relation: parent
+    object: organization:EcoLIfe
+  - user: user:Jane
+    relation: viewer
+    object: organization:GlobalCorp
+  - user: user:Jane
+    relation: editor
+    object:  organization:EcoLIfe
+
+tests:
+  - name: Jane has limited access
+    check:
+      - user: user:Jane
+        object: organization:GlobalCorp
+        assertions:
+          can_view: true
+          can_edit: false
+      - user: user:Jane
+        object: organization:EcoLIfe
+        assertions:
+          can_view: true
+          can_edit: true
+
+```
+
+To execute the tests, you utilize the OpenFGA CLI with the following command:
+
+```bash
+fga model test --tests .\tests.fga.yml
+```
+
+These tests explicitly verify whether the authorization model correctly interprets Jane's permissions as intended. 
+By running these tests, teams can proactively identify and rectify issues, ensuring the model's accuracy before it impacts the live environment.
+
+This structured approach to defining and testing authorization models within OpenFGA not only streamlines development workflows but also significantly boosts confidence in the security and correctness of the implemented permissions system.
+
+## OpenFGA Server: Configuration and Operations
+
+OpenFGA introduces a more dynamic and flexible approach to managing authorization models compared to Ory Keto. 
+While Ory Keto integrates the authorization model as a fixed configuration during startup, 
+OpenFGA enhances operational flexibility by supporting multiple authorization models, each housed within its own distinct storage environment, referred to as a `store`.
+
+A store in OpenFGA acts as an encapsulation layer, designed to segment and manage different sets of authorization models. 
+This architecture is particularly advantageous for organizations that operate across multiple teams or development stages. 
+Each store can contain several authorization models, allowing for tailored configurations suited to specific team requirements or project phases.
+
+One of the critical advantages of this setup is the ability to version authorization models effectively. 
+For instance, when a new model version is rolled out, it does not necessitate an immediate migration by all services. 
+Instead, services can continue to operate using older, well-tested versions of the model until they are ready to transition. 
+This capability is crucial for maintaining stability and continuity in large-scale environments where sudden changes can disrupt operations.
+
```

Comment:
it also allows shadow testing (forgot how exactly it is called):

for a few transitional weeks, for each request, run both systems in production, use the old value, but check if the new response would be the same. If it differs, you might have a problem